File: packages/contracts/policy.py
from typing import List, Optional
from pydantic import BaseModel, Field

class RecommendationData(BaseModel):
    recommendation_id: str
    hotspot_id: str
    evidence_bundle_id: str
    title: str
    problem: str
    proposed_intervention: str
    intended_beneficiaries: int
    supporting_evidence_ids: List[str] = Field(default_factory=list)
    risks: List[str] = Field(default_factory=list)
    missing_information: List[str] = Field(default_factory=list)
    confidence: float
    status: str = "under_review"
    ai_draft: bool = True
    human_approved: bool = False
    schema_version: str = "recommendation-1.0.0"

class PolicyDecisionData(BaseModel):
    decision_id: str
    recommendation_id: str
    action: str = Field(..., description="approve_for_assessment, request_evidence, edit, assign, defer, reject")
    reason: str
    actor_id: str
    actor_role: str = "decision_maker"
    decided_at: str

class ImpactMetricData(BaseModel):
    project_id: str
    metric_code: str
    baseline: float
    target: float
    current: float
    unit: str
    source_id: str
    measured_at: str
    confidence: float

File: frontend/src/lib/api/policy.ts
import { z } from "zod";
import { apiRequest } from "./client";
import { metricSchema, projectSchema, recommendationSchema } from "./schemas";

export const policyKeys = { recommendations: ["policy", "recommendations"] as const, projects: ["policy", "projects"] as const, metrics: (id: string) => ["policy", "projects", id, "metrics"] as const };
export const policyApi = {
  recommendations: () => apiRequest("/api/policy/recommendations", z.array(recommendationSchema)),
  createRecommendation: (input: { hotspot_id: string; evidence_bundle_id: string; title?: string }) => apiRequest("/api/policy/recommendations", recommendationSchema, { method: "POST", body: JSON.stringify(input) }),
  decide: (id: string, input: { action: "approve_for_assessment" | "request_evidence" | "defer" | "reject"; reason: string; actor_id: string; actor_role: string }) => apiRequest(`/api/policy/recommendations/${encodeURIComponent(id)}/decisions`, z.record(z.string(), z.unknown()), { method: "POST", body: JSON.stringify(input) }),
  projects: () => apiRequest("/api/policy/projects", z.array(projectSchema)),
  createProject: (input: { recommendation_id: string; title?: string; assigned_department?: string }) => apiRequest("/api/policy/projects", projectSchema, { method: "POST", body: JSON.stringify(input) }),
  metrics: (id: string) => apiRequest(`/api/policy/projects/${encodeURIComponent(id)}/metrics`, z.array(metricSchema)),
  addMetric: (id: string, input: { metric_code: string; baseline: number; target: number; current: number; unit: string; source_id: string; measured_at?: string; confidence: number }) => apiRequest(`/api/policy/projects/${encodeURIComponent(id)}/metrics`, metricSchema, { method: "POST", body: JSON.stringify(input) }),
};
